pkg/logql/syntax/ast.go

```diff
@@ -302,10 +314,17 @@ func (e *LineFilterExpr) Filter() (log.Filterer, error) {
 			}
 			acc = append(acc, next)
 		default:
-			next, err := log.NewFilter(curr.Match, curr.Ty)
+			var next log.Filterer
```

Comment:
Actually generate the filters

Reply:
Why is this required? Shouldn't the parsing already take care or creating the right filter? I'm a little confused by this?

Reply:
There may be a better way to do this. Afaict, the parser is responsible for creating a `*LineFilterExpr` but we don't get actual filters until we call `LineFilterExpr.Filter()` If the parser creates our type, it won't be composable with other `LineFilterExprs` because that's a concrete type rather than and interface.